File: src/com/glureau/kotlin/sandbox/adapter_when/java/MouseMonkey.java
package com.glureau.kotlin.sandbox.adapter_when.java;

import java.util.Random;
import java.util.Timer;
import java.util.TimerTask;

/**
 * Random inputs
 */
public class MouseMonkey {

    private final Timer timer;

    public MouseMonkey(final MouseListener listener) {
        timer = new Timer();
        timer.scheduleAtFixedRate(new TimerTask() {
            @Override
            public void run() {
                int rand = new Random().nextInt(5);
                switch (rand) {
                    case 0:
                        listener.mouseClicked(new MouseEvent());
                        break;
                    case 1:
                        listener.mouseEntered(new MouseEvent());
                        break;
                    case 2:
                        listener.mouseExited(new MouseEvent());
                        break;
                    case 3:
                        listener.mousePressed(new MouseEvent());
                        break;
                    case 4:
                        listener.mouseReleased(new MouseEvent());
                        break;
                }
            }
        }, 0, 100);
    }

    public void stop() {
        timer.cancel();
    }
}

File: src/com/glureau/kotlin/sandbox/adapter_when/kt/MouseMonkey.kt
package com.glureau.kotlin.sandbox.adapter_when.kt;

import java.util.*


/**
 * Random inputs
 */
class MouseMonkey(listener: MouseListener) {

    private val timer: Timer

    init {
        timer = Timer()
        timer.scheduleAtFixedRate(object : TimerTask() {
            override fun run() {
                val rand = Random().nextInt(5)
                when (rand) {
                    0 -> listener.mouseClicked(MouseEvent())
                    1 -> listener.mouseEntered(MouseEvent())
                    2 -> listener.mouseExited(MouseEvent())
                    3 -> listener.mousePressed(MouseEvent())
                    4 -> listener.mouseReleased(MouseEvent())
                }
            }
        }, 0, 100)
    }

    fun stop() {
        timer.cancel()
    }
}
